### ISSUE TITLE
Comments appearing on different threads, in different groups

### ISSUE THREAD

**Issue by strypey** [loomio/loomio#9814]
I just saw comments that I definitely made on one thread, appearing on a different thread, in a completely different group. In one case, a very personal comment made in a private group, was posted in a very public group. I don't know if this was only a failure of the web client on my end, or whether anyone viewing those threads would have seen the decontextualised comments. If they can, I have no idea what other comments may have been affected. This is very disturbing!

**Comment by robguthrie** [comment]
Thanks for reporting it. Was the comment on the end of the thread, or a
reply?

**Comment by strypey** [comment]
@robguthrie:
> Thanks for reporting it. Was the comment on the end of the thread, or a reply?

They were both replies.

**Comment by strypey** [comment]
FYI this has happened  to least one other member in a private group I'm in. I've shared the link to this issue with the members of that group, and encouraged them to add their reports here. I doubt she's using the kind of unusual OS/ browser combo I am, which suggests the problem is with Loomio itself.

All the affected content seems to be replies to other comments, so perhaps the problem is in the code that implements comment threading?

**Comment by robguthrie** [comment]
Hi @strypey I think I've found the bug.

The steps to reproduce it were
- open a thread
- change to another thread via the notifications dropdown menu
- write a comment in the comment box at the bottom of the thread and click send. 
- the comment appears on the previous, rather than current thread

To be clear. I don't think this bug was happening when you click "Reply" and write a comment somewhere in the middle of the thread. 

Can you recall if you were "replying" (in the middle of the thread) or "adding comment" (usually at the bottom of the thread)?

**Comment by strypey** [comment]
@robguthrie I note the little message on Loomio saying the problem has been fixed. Great work! I'll be sure to let you know if I spot the bug in the wild again, now that this fix has been applied. Just for the record...

> Can you recall if you were "replying" (in the middle of the thread)

Yes, that's the only time I've noticed the behaviour. It makes sense to me that the problem would be related to...

> change to another thread via the notifications dropdown menu

... maybe after being auto-navigated to a specific comment via clicking on a notification.

BTW The member in the private Whirlpool group, who also noticed the problem, pointed out that it would be inappropriate to share screenshots from our group illustrating the problem, on a public thread (ie this Issue). They seem shy in general about posting their experiences on a public thread. Is there a private support channel I could direct them to?

**Comment by robguthrie** [comment]
I'm concerned that I've not found a problem with the "reply" action, only the "new comment" action. 

The main (and private) channel for bug reports or feedback is https://www.loomio.com/contact or contact@loomio.org, which you can find in your sidebar menu.

**Comment by strypey** [comment]
@robguthrie
> I'm concerned that I've not found a problem with the "reply" action, only the "new comment" action

I just ran a quick test, but couldn't reproduce the bug. The steps I followed were

* open a thread to a specific comment, using the notifications menu
 
* change to a comment in another thread, using the notifications menu

* click on 'reply', write a comment and click send

I don't want to celebrate prematurely but I think you've squashed it. I'll keep my eyes peeled though.
